### PR TITLE
PORTALS-2157 - Rename facetAliases to columnAliases + related cleanup

### PR DESCRIPTION
**Pull request by nickgros**
- Rename facetAliases to columnAliases (since it applies to all columns, regardless of whether they are faceted)
- columnAliases are only used within QueryVisualizationWrapper
- QueryVisualizationWrapper now passes `getColumnDisplayName` through context
- Alias consumers now use `getColumnDisplayName`
- Refactor unrelated logic from unCamelCase to QueryVisualizationWrapper

### CODE REVIEW

File: src/__tests__/lib/containers/QueryVisualizationWrapper.test.tsx [hunk 1]
```diff
@@ -228,4 +234,49 @@ describe('QueryVisualizationWrapper', () => {
     // The array of visible columns should not have changed, and should have referential equality
     expect(firstSetOfVisibleColumns).toBe(secondSetOfVisibleColumns)
   })
+
+  describe('getColumnDisplayName', () => {
```
Comment (nickgros) [inline src/__tests__/lib/containers/QueryVisualizationWrapper.test.tsx:238]: New tests to verify existing behavior

File: src/lib/containers/CardContainer.tsx [hunk 2]
```diff
@@ -47,14 +47,23 @@ export const CardContainer = (props: CardContainerProps) => {
     infiniteQueryContext
 
   const queryRequest = getLastQueryRequest()
+
+  const queryVisualizationContext = useQueryVisualizationContext()
+
   const renderCard = (props: any, type: string) => {
```
Comment (nickgros) [inline src/lib/containers/CardContainer.tsx:53]: TODO: fn component defined in a fn component => excessive rerenders

File: src/lib/containers/CardContainerLogic.tsx [hunk 3]
```diff
@@ -150,7 +153,7 @@ export const CardContainerLogic = (props: CardContainerLogicProps) => {
     concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
     entityId: parseEntityIdFromSqlStatement(sql),
     query: {
-      sql: sql,
+      sql: props.sql,
```
Comment (nickgros) [inline src/lib/containers/CardContainerLogic.tsx:156]:
```suggestion
      sql: sql,
```

File: src/lib/containers/CardContainer.tsx [hunk 4]
```diff
@@ -27,11 +28,26 @@ export type CardContainerProps = {
   isHeader?: boolean
   isAlignToLeftNav?: boolean
   title?: string
-  facetAliases?: Record<string, string>
   isLoading?: boolean
   unitDescription?: string
 } & CardConfiguration
 
+function Card(props: { propsToPass: any; type: string }) {
```
Comment (nickgros) [inline src/lib/containers/CardContainer.tsx:35]: This was defined in the CardContainer component below, which probably resulted in excessive re-renders

File: src/__tests__/lib/containers/GenericCard.test.tsx [hunk 5]
```diff
@@ -29,11 +29,20 @@ import {
...
 const renderComponent = (props: GenericCardProps) => {
-  return render(<GenericCard {...props} />, {
-    wrapper: createWrapper(),
-  })
+  return render(
+    <GenericCard
+      {...props}
+      queryVisualizationContext={{
+        getColumnDisplayName: jest.fn().mockImplementation(col => col),
+      }}
```
Comment (nickgros) [inline src/__tests__/lib/containers/GenericCard.test.tsx:40]: Many of the test changes are mocking the new `getColumnDisplayName` function

File: src/lib/containers/QueryVisualizationWrapper.tsx [hunk 6]
```diff
@@ -133,18 +143,34 @@ export function QueryVisualizationWrapper(
     )
   }, [selectColumns, lastQueryRequest.query.sql, props.visibleColumnCount])
 
+  const getColumnDisplayName = useCallback(
+    (columnName?: string) => {
+      // SWC-5982: if force-display-original-column-names is set, then just return the string
+      const forceDisplayOriginalColumnName =
+        localStorage.getItem('force-display-original-column-names') === 'true'
```
Comment (nickgros) [inline src/lib/containers/QueryVisualizationWrapper.tsx:150]: This could easily be changed to a prop now

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**